src/skadi/sample.cc

```diff
@@ -456,9 +443,8 @@ template <class coord_t> double sample::get(const coord_t& coord) {
 
   // get the proper source of the data
   const auto& tile = cache_->source(index);
```

Comment:
When we fetch tile - we do cache insert with locked mutex. Need to double check if we should lock here as well...

Reply:
i don't see why a lock is needed here. it seems that the state of the cache is guarded inside of the source method